File: Fibonacci.java
class Fibonacci
{
	public static int fibonacci(int n) {
		return n < 2 ? n : fibonacci(n - 2) + fibonacci(n - 1);
	}

	public static void main(String[] args) {
		System.out.println(fibonacci(45));
		//fibonacci(40);
	}
}

File: fibonacci.go
/**
 * User: guoyao
 * Time: 06-20-2013 12:16
 * Blog: http://www.guoyao.me
 */

package main

import "fmt"

func main() {
	//fmt.Println(fibonacci(40))
	fmt.Println(fibonacci(45))
}

func fibonacci(n uint) (result uint) {
	if n < 2 {
		return n
	}
	return fibonacci(n-2) + fibonacci(n-1)
}
